fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserProperty.java

```diff
@@ -121,6 +126,17 @@ public long getMaxQueryInstances() {
         return commonProperties.getMaxQueryInstances();// maxQueryInstances;
     }
 
+    public String[] getSqlBlockRules() {
+        if (this.sqlBlockRulesSplit.length != 0) {
+            return this.sqlBlockRulesSplit;
+        }
+        String sqlBlockRules = commonProperties.getSqlBlockRules();
```

Comment:
This method is only protected by `readLock`, but you do write operation here.
Why not just save the splitted rules in `commonProperties`?